metrics/cloudwatch/cloudwatch.go

```diff
@@ -133,11 +144,41 @@ func (cw *CloudWatch) Send() error {
 		}
 	}
 
-	_, err := cw.svc.PutMetricData(&cloudwatch.PutMetricDataInput{
-		Namespace:  aws.String(cw.namespace),
-		MetricData: datums,
-	})
-	return err
+	var tokens = make(chan struct{}, cw.numConcurrentRequests)
+	var errors = make(chan error)
+	var n int
+
+	for len(datums) > 0 {
+		var batch []*cloudwatch.MetricDatum
+		lim := min(len(datums), cw.numConcurrentRequests)
+		batch, datums = datums[:lim], datums[lim:]
+		n++
+		go func(batch []*cloudwatch.MetricDatum) {
+			tokens <- struct{}{}
+			_, err := cw.svc.PutMetricData(&cloudwatch.PutMetricDataInput{
+				Namespace:  aws.String(cw.namespace),
+				MetricData: batch,
+			})
+			<-tokens
+			errors <- err
+		}(batch)
```

Comment:
Since you construct the batch within the loop scope, there's no reason to pass it as a parameter to this anonymous function.